wrangler-transform/src/e2e-test/java/io/cdap/plugin/wrangler/actions/ValidationHelper.java

```diff
@@ -0,0 +1,120 @@
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Validation Helper.
+ */
+public class ValidationHelper {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ValidationHelper.class);
+  static Gson gson = new Gson();
+  public static boolean validateActualDataToExpectedData(String table, String fileName) throws IOException,
+    InterruptedException, URISyntaxException {
+    Map<String, JsonObject> bigQueryMap = new HashMap<>();
+    Map<String, JsonObject> fileMap = new HashMap<>();
+    Path importExpectedFile = Paths.get(ValidationHelper.class.getResource("/" + fileName).toURI());
+
+    getBigQueryTableData(table, bigQueryMap);
+    getFileData(importExpectedFile.toString(), fileMap);
+
+    boolean isMatched = bigQueryMap.equals(fileMap);
+
+    return isMatched;
+  }
+
+  public static void getFileData(String fileName, Map<String, JsonObject> fileMap) {
+    try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        JsonObject json = gson.fromJson(line, JsonObject.class);
+        String idKey = getIdKey(json);
+        if (idKey != null) {
+          JsonElement idElement = json.get(idKey);
+          if (idElement.isJsonPrimitive()) {
+            String idValue = idElement.getAsString();
+            fileMap.put(idValue, json);
+          }
+        } else {
+          Log.error("ID key not found");
+        }
+      }
+    } catch (IOException e) {
+      System.err.println("Error reading the file: " + e.getMessage());
+    }
+  }
+
+  private static void getBigQueryTableData(String targetTable, Map<String, JsonObject> bigQueryMap)
+    throws IOException, InterruptedException {
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + targetTable + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+
+    for (FieldValueList row : result.iterateAll()) {
+      JsonObject json = gson.fromJson(row.get(0).getStringValue(), JsonObject.class);
+      String idKey = getIdKey(json); // Get the actual ID key from the JSON object
+      if (idKey != null) {
+        JsonElement idElement = json.get(idKey);
+        if (idElement.isJsonPrimitive()) {
+          String idk = idElement.getAsString();
+          bigQueryMap.put(idk, json);
+        } else {
+          LOG.error("Data Mismatched");
+        }
+      }
+    }
+  }
+
+  /**
+   * Retrieves the key for the ID element in the provided JSON object.
+   *
+   * @param json The JSON object to search for the ID key.
+   */
+  private static String getIdKey(JsonObject json) {
```

Comment:
We can remove the need for this complicated id checking logic by ensuring that in the test data that we use, there is always an 'id' (case-insensitive name) column.